backend/app/services/implementations/auth_service.py

```
@@ -219,3 +219,13 @@ def is_authorized_by_email(self, access_token, requested_email):
             )
         except:
             return False
+        
+    def is_authorized_by_token(self, access_token):
+        try:
+            decoded_id_token = firebase_admin.auth.verify_id_token(
+                access_token, check_revoked=True
+            )
+            firebase_user = firebase_admin.auth.get_user(decoded_id_token["uid"])
+            return firebase_user.email_verified
+        except:
+            return False
```

So one q here ; this function returns a `True` / `False`, but when there's an exception it by default returns false. This, however, will not cause the exception to be caught in the calling sites (for instance, in the route calling `is_verified`, if an exception is raised within `is_authorized_by_token`, we will never see it propagate up, and thus will return a 200 still). Just wanted to know if this was intentional ; otherwise, I think we should introduce some custom exceptions to account for the error down here, and throw that instead of returning `False` by default

---

I think we should at least console out an error here (as a temp soln) just so we have some visibility around this in our logs.